scheduler/conflicts.py

```diff
@@ -0,0 +1,62 @@
+"""module responsible fro finding conflicts in database"""
+from typing import List, Tuple
+from enum import Enum
+from scheduler.models import Lesson, Professor, Auditorium, Group
+
+class ConflictType(Enum):
+    PROFESSOR = 1
+    AUDITORIUM = 2
+    GROUP = 3
+
+def check_time_conflict(lesson1: Lesson, lesson2: Lesson) -> bool:
+    """
+    Check if lesson2 started or ended during lesson1.
+    Lessons should have common factor like same professor, auditorium, group
+    :param lesson1, lesson2: Lesson object retrived from database
+    :return: bool if lessons are conflicting
+    """
+    if lesson1.start_time <= lesson2.start_time and lesson1.end_time > lesson2.start_time:
```

Comment:
Nie ma przypadków kiedy lekcja 1(2) w całości zawiera się w lekcji 2(1)

Reply:
Faktycznie gdyby pierwsza lekcja zawierała się całkowicie w drugiej nie widziałoby problemu, dodałem nowy warunek. Gdy 2 zawiera się w 1 powinno być okej, bo oznacza, że zacznie/skończy się w trakcie pierwszej i wyłapią to wcześniejsze warunki